### PR TITLE
Setting-app-testing

### PR DESCRIPTION
**Pull request by Perumal605**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208510368102329

### CODE REVIEW

File: tests/SettingAppTest.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+     */
+    public function it_raw()
+    {
+        $user_apps_services = container()->get( SettingsApps::class );
+        $apps = [ app_factory(), app_factory() ];
+
+        $save = $user_apps_services->save( $apps );
+        $setting_apps = container()->get( SettingsApps::class )->raw();
+
+        $this->assertCount( 2, $setting_apps );
```
Comment (incraigulous) [inline tests/SettingAppTest.php:23]: @Perumal605 , please loop though each $app and make sure it's slug exists in the database. You can use `->find()` for this, or search the $settings_apps array.

File: tests/SettingAppTest.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        $apps = app_factory( [ 'is_deleted' => true ] );
+        $setting_apps = container()->get( SettingsApps::class )->is_allowed( $apps );
+
+        $this->assertFalse( $setting_apps );
+    }
+
+    /**
+     * @test
+     */
+    public function it_save()
```
Comment (incraigulous) [inline tests/SettingAppTest.php:41]: Rename to `it_saves`.

File: tests/SettingAppTest.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        $settings_apps_service = container()->get( SettingsApps::class );
+
+        $settings_apps_service->save( $apps );
+        $app_slugs = array_column( $apps, 'slug' );
+
+        $get_apps = container()->get( SettingsApps::class )->raw();
+        $get_app_slugs = array_column( $get_apps, 'slug' );
+
+        foreach ( $app_slugs as $slug ) {
+            $this->assertContains( $slug, $get_app_slugs );
```
Comment (incraigulous) [inline tests/SettingAppTest.php:53]: Great!

File: tests/SettingAppTest.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        }
+        $setting_apps = $settings_apps_service->raw();
+        $this->assertCount( 2, $setting_apps );
+    }
```
Comment (incraigulous) [inline tests/SettingAppTest.php:57]: Please add the remaining methods that are inherited from `AppSource` and `Source`

**Review by incraigulous — CHANGES_REQUESTED** [review]

**Comment by Perumal605** [comment]
@incraigulous I have pushed the latest updates here.

**Pull request merged**